packages/next-server/lib/router/router.js

```diff
@@ -291,7 +291,7 @@ export default class Router {
         return { error: err }
       }
 
-      const Component = this.ErrorComponent
```

Comment:
The `this.ErrorComponent` property is still getting destructured and set by the constructor if there aren't any dependencies relying on its existence we could remove this in addition to the branch that guards against `ErrorComponent` in the constructor L20.